### DISCUSSION TITLE
"Unknown" Device name while parsing a particular user agent

### DISCUSSION THREAD

**Discussion by AdarshPanigrahi** [nielsbasjes/yauaa#1050] [Q&A] [answered]
Hi
While parsing a user agent, (Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Mobile Safari/537.36) the device name is coming out as unknown. Whereas for similar useragents (Mozilla/5.0 (Linux; Android 10; P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Mobile Safari/537.36) the device name is showing as "Android Mobile".
Is this a regular behaviour or a missed case?

**Comment by nielsbasjes** [comment]
Yes, that is indeed inconsistent. I'll have a look.

**Reply by AdarshPanigrahi** [reply]
Can this particular corner case be fixed in a future release ?

**Reply by nielsbasjes** [reply]
Yes, I have the fix done locally.

**Reply by nielsbasjes** [reply, marked as answer]
Fixed in 7.19.2
https://try.yauaa.basjes.nl/?ua=Mozilla%2F5.0+%28Linux%3B+Android+10%3B+K%29+AppleWebKit%2F537.36+%28KHTML%2C+like+Gecko%29+Chrome%2F113.0.0.0+Mobile+Safari%2F537.36

**Reply by JeffHu88** [reply]
Hi @nielsbasjes , Has this issue been fixed? I'm using the latest 7.31.0 version, and the OperatingSystemVersion still shows "??".

**Reply by nielsbasjes** [reply]
In the above example there is currently nothing else to fix.
The "??" is the correct result.

**Reply by JeffHu88** [reply]
Isn't the Operating System's Version Android 10?

**Reply by nielsbasjes** [reply]
No, the 10 you see is a lie.

Read this explanation 
https://yauaa.basjes.nl/developer/reportingissues/index.html#these-are-not-bugs

**Reply by JeffHu88** [reply]
ok! I see. thanks a lot.
